### PR TITLE
fix(libp2p): expose builder phase error

### LINKED ISSUE #4829: Make `libp2p::builder::phase::behaviour::BehaviourError` importable
### Description

Currently there's no way to import `libp2p::builder::phase::behaviour::BehaviourError` as it's inside a private module. 
But it's being exposed in a public interface

### Motivation

No way to use it in user-code without boxing it (again)

### Current Implementation

Not publicly exposed

### Are you planning to do it yourself in a pull request ?

No

### PR DESCRIPTION
**Pull request by drHuangMHT**
## Description
May close #4829 and #4824.  
Export three error types `BehaviourError`, `TransportError`, `WebsocketError` with rename. Feature gated `WebsocketError`.  
Exported at crate root as [suggested](https://github.com/libp2p/rust-libp2p/issues/4824#issuecomment-1803013514).

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: libp2p/src/builder/phase.rs [hunk 1]
```diff
@@ -16,17 +16,21 @@ mod websocket;
...
 use libp2p_identity::Keypair;
+pub use other_transport::TransportError;
 use other_transport::*;
 use provider::*;
 use quic::*;
 use relay::*;
 use swarm::*;
 use tcp::*;
+#[cfg(all(not(target_arch = "wasm32"), feature = "websocket"))]
+pub use websocket::WebsocketError;
```
Comment (elenaf9) [inline libp2p/src/builder/phase.rs]: Nit: can we order public items below the private ones?

**Review by elenaf9 — APPROVED** [review]
LGTM. Thanks @drHuangMHT!

File: libp2p/src/builder/phase.rs (hunk 1 above)
Comment (drHuangMHT) [reply libp2p/src/builder/phase.rs]: I would love to but it seems that `cargo fmt` won't let me. It always reorder them. And I believe the formatting test won't pass.

Comment (elenaf9) [reply libp2p/src/builder/phase.rs]: Even when leaving one empty line between the two blocks?

Comment (drHuangMHT) [reply libp2p/src/builder/phase.rs]: I've tried, the answer is no for some reason.  
EDIT: I believe it has something to do with my toolchain. I'll try pushing it regardless to see if the test passes

**Comment by drHuangMHT** [comment]
cc @jxs

**Review by jxs — COMMENTED** [review]
thanks @drHuangMHT! Looks good to me, I just feel this should be a `fix` instead, and we should add an entry to the `CHANGELOG.md` files wdyt?

File: libp2p/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## unreleased
```
Comment (jxs) [inline libp2p/CHANGELOG.md]: `0.54.2` hasn't been released yet, you can put the changelog entry below it :P

**Review by jxs — APPROVED** [review]
Thanks!

**Pull request merged**